analyses/infercnv-consensus-cell-type/template-notebooks/ewings-infercnv-results.Rmd

```diff
@@ -0,0 +1,536 @@
+
+ewings_violin <- ggplot(query_has_cnv_df) +
+  aes(
+    x = ewing_annotation,
+    y = total_cnv_per_cell,
+    fill = normal_reference
+  ) +
+  geom_violin(scale = "width") +
+  scale_fill_brewer(palette = "Pastel2") +
+  labs(title = "Annotations from cell-type-ewings") +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+
+wrap_plots(consensus_violin, ewings_violin, ncol = 1, guides = "collect") +
+  plot_annotation(title = "CNV scores across cell types") &
+  theme(legend.position = "bottom")
+```
+
+
+
+## UMAPs
+
+Next, we'll visualize the total CNV counts as a UMAP.
+
+First, as a guide to interpret the UMAP, let's color by the top 10 `cell-type-ewings` cell types.
+
+```{r, fig.width = 6}
+ggplot(umap_df) +
+  aes(x = UMAP1, y = UMAP2, color = forcats::fct_lump(ewing_annotation, 8)) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  labs(color = "ewings annotation") +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1))) +
+  umap_theme
+```
+
+Next, the total CNVs per cell; ideally, higher values will overlap with regions in the UMAP containing putative tumor cells.
+
+```{r fig.width=10, warning = FALSE}
+# noting we expect warnings with this plot since there's no UMAP coordinates
+# for cells from other libraries in the reference
+ggplot(has_cnv_df) +
+  aes(x = UMAP1, y = UMAP2, color = total_cnv_per_cell) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  scale_color_viridis_c() +
+  facet_wrap(vars(normal_reference)) +
+  labs(title = "Total CNV per cell") +
+  umap_theme
+```
+
+## Compare references directly
+
+We can also look at CNV predictions directly between references using a point-density scatterplot, with the Spearman correlation shown:
+
+```{r, fig.height=6}
+compare_refs <- has_cnv_df |>
+  dplyr::filter(library_id == params$library_id) |>
+  dplyr::select(
+    barcodes, normal_reference, total_cnv_per_cell
+  ) |>
+  tidyr::pivot_wider(
+    names_from = normal_reference,
+    values_from = total_cnv_per_cell
+  ) |>
+  # color points this way rather than binning to ensure small bins aren't left out
+  dplyr::add_count(`all-immune`, `subset-immune`)
+
+# Determine max limit for axes
+max_axis <- max(has_cnv_df$total_cnv_per_cell)
+axis_breaks <- 1:max_axis
+
+# Calculate in spearman correlation coefficient to include in plot
+rho <- cor(compare_refs$`all-immune`, compare_refs$`subset-immune`, method = "spearman") |>
+  round(2)
+
+ggplot(compare_refs) +
+  aes(
+    x = `all-immune`,
+    y = `subset-immune`,
+    color = n
+  ) +
+  geom_point(size = 5) +
+  geom_abline(color = "red") +
+  scale_color_viridis_c() +
+  coord_equal() +
+  scale_x_continuous(limits = c(0, max_axis), breaks = 1:max_axis) +
+  scale_y_continuous(limits = c(0, max_axis), breaks = 1:max_axis) +
+  labs(
+    title = "Per-cell CNVs between normal references",
+    subtitle = glue::glue("rho = {rho}"),
+    color = "Number of cells"
+  ) +
+  theme(legend.position = "bottom")
+```
+
+## Clusters
```

Comment:
For all the plots where you have clusters as the x-axis can you order the clusters so the numbers are in the right order? right now s1, s10, and s11 are all in order.